docs/src/main/asciidoc/building-native-image.adoc

```diff
@@ -623,6 +624,81 @@ Quarkus provides the `quay.io/quarkus/quarkus-distroless-image:1.0` image.
 It contains the required packages to run a native executable and is only **9Mb**.
 Just add your application on top of this image, and you will get a tiny container image.
 
+=== Separating Java- and native-image compilation
+
+In certain circumstances, you may want to separate compilation of the Java and native-image.
+For example, in a CI/CD pipeline, you may want to have one step to generate the JAR file and another step to compile this JAR file into the native executable.
+For this use case, you can set the `quarkus.package.type=native-sources`.
+This will execute the java compilation as if you would have started native compilation (`-Pnative`), but stop before triggering the actual call to GraalVM's `native-image`.
```

Comment:
```suggestion
This will execute the java compilation as if you would have started native compilation (`-Pnative`), but stops before triggering the actual call to GraalVM's `native-image`.
```